Developing-new-features-with-feature-gating.md

```diff
@@ -145,6 +145,91 @@ spyOnProperty(platformFeatureService, 'status', 'get').and.returnValue(
 );
 ```
 
+#### Backend
+
+For example, if you wish to write a test with the required feature flag enabled, you can do so by following the steps below (the approach to testing with the feature flag disabled is the same up to step 4):
+
+1. Import the following modules into the test file:
+
+```python
+from core.domain import platform_feature_services as feature_services # (alias is optional)
+from core.domain import platform_parameter_domain
+from core.domain import platform_parameter_list
+from core.domain import platform_parameter_registry as registry # (alias is optional)
+```
+
+2. Clear the parameter registry and create a mock feature flag (using the actual enum of the feature flag -- we recommend you do it in the `setUp` hook), like so:
+
+```python
+self.mock_feature_flag = registry.Registry.create_feature_flag(
+  platform_parameter_list.ParamNames.NewFeature, 'a mock of the new_feature feature flag',
+  platform_parameter_domain.FeatureStages.DEV) # FeatureStage DEV just simplifies things
+```
+
+3. Set the rules for when the feature flag is to be enabled (in the method which performs the testing), like so (here we're setting the flag to be enabled when in DEV mode):
+
+```python
+test_can_do_something(self) -> None:
+  feature_services.update_feature_flag_rules(
+            platform_parameter_list.ParamNames.NewFeature.value, self.owner_id, 'test update', # owner_id is the id of the user that is updating the feature flag. See note underneath this code block.
+            [
+                platform_parameter_domain.PlatformParameterRule.from_dict({
+                    'filters': [
+                        {
+                            'type': 'server_mode',
+                            'conditions': [
+                                ['=', platform_parameter_domain.FeatureStages.DEV.value]
+                            ]
+                        }
+                    ],
+                    'value_when_matched': True
+                })
+            ]
+        )
+
+```
+
+**NOTE**: Like mentioned above `self.owner_id` refers to the id of the user with the necessary permissions to perform the actual feature flag rules update. Some tests have the `owner_id` property created within the `setUp()` hook, and this variable can be used throughout the test class.
```

Comment:
Can you explain what permissions owner_id must have and how to make sure they have them? (They will need superadmin permissions, right?)

Reply:
Added that the code block shows how to access the owner account credentials, and that they provide super admin permissions to an account created using them, which are used to update the feature flag rules.

> how to make sure they have them

An account made using the `self.OWNER_EMAIL, self.OWNER_USERNAME` credentials, which are a part of the `GenericTestBase` class and are available to every test class, has those permissions. To make sure the created account has these permissions, the developer needs to use those credentials to create the account.